src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.Helpers.cs

```diff
@@ -31,6 +31,7 @@ public static partial class JsonSerializer
             var reader = new Utf8JsonReader(utf8Json, isFinalBlock: true, readerState);
 
             ReadStack state = default;
+            jsonTypeInfo.EnsureConfigured();
```

Comment:
Could this be moved to the JsonTypeInfo construction phase (to avoid making contention a concern?)

Reply:
No for 3 reasons:
- this code path can be hit from APIs taking directly JsonTypeInfo which could not be preconfigured
- putting that directly in JsonTypeInfo constructor can cause stack overflow for source gen
- type returned from GetTypeInfo needs to not congiured as it needs to allow for editing as it's a common scenario. Configuring produces the cache

Reply:
Got it, essentially it is accounting for instances not returned from the `JsonSerializerOptions` cache so cycles cannot be accounted for?

Reply:
I'm not sure what do you mean by `cycles` here. But yes for the first part of the question - this accounts for source gen JsonTypeInfos